libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -98,29 +76,25 @@ private data class DynamicallyUpdatedRouteValues(
 - navigation.depart
 - navigation.feedback
 - navigation.reroute
-- navigation.fasterRoute
 - navigation.arrive
 - navigation.cancel
 The class must be initialized before any telemetry events are reported. Attempting to use telemetry before initialization is called will throw an exception. Initialization may be called multiple times, the call is idempotent.
 The class has two public methods, postUserFeedback() and initialize().
  */
-@SuppressLint("StaticFieldLeak")
-internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
-    internal const val LOCATION_BUFFER_MAX_SIZE = 20
-    private const val ONE_SECOND = 1000
-    private const val MOCK_PROVIDER =
-        "com.mapbox.navigation.core.replay.ReplayLocationEngine"
+internal object MapboxNavigationTelemetry :
+    RouteProgressObserver,
+    RoutesObserver,
+    OffRouteObserver,
+    NavigationSessionStateObserver {
+    internal val TAG = Tag("MAPBOX_TELEMETRY")
 
-    internal const val TAG = "MAPBOX_TELEMETRY"
+    private const val ONE_SECOND = 1000
```

Comment:
Should this be int or floating-point? Can the rounding down of calculations impact the system or telemetry schema?

Reply:
`secondsSinceLastReroute` is `Int` as in the schema. I believe we should be good.

Reply:
Got it, thanks for confirming :+1: